dirty_cat/datetime_encoder.py

```diff
@@ -0,0 +1,171 @@
+        elif feature == "second":
+            return pd.DatetimeIndex(date_series).second.to_numpy()
+        elif feature == "millisecond":
+            return pd.DatetimeIndex(date_series).millisecond.to_numpy()
+        elif feature == "microsecond":
+            return pd.DatetimeIndex(date_series).microsecond.to_numpy()
+        elif feature == "nanosecond":
+            return pd.DatetimeIndex(date_series).nanosecond.to_numpy()
+        elif feature == "dayofweek":
+            return pd.DatetimeIndex(date_series).dayofweek.to_numpy()
+        elif feature == "holiday":
+            # Create an indicator for holidays
+            cal = calendar()
+            holidays = cal.holidays(start=date_series.min(), end=date_series.max())
+            return np.isin(date_series, holidays).astype(int)
+        elif feature == "other":
+            # Gather all the variables below the extract_until into one numerical variable
+            res = (pd.to_datetime(date_series) - pd.to_datetime(pd.DatetimeIndex(date_series).floor(
+                self.word_to_alias[self.extract_until]))).to_numpy()
+            # Convert to the extract_until unit (e.g if I extract until "minute", then convert to minutes)
+            return res / pd.to_timedelta(1, self.word_to_alias[self.extract_until])
+
+    def fit(self, X, y=None):
+        """
+        Fit the DatetimeEncoder to X. In practice, just stores which extracted features
+        are not constant.
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            Data where each column is a datetime feature.
+        Returns
+        -------
+        self
+            Fitted DatetimeEncoder instance.
+        """
+        self._validate_keywords()
+        self.to_extract_full = self.to_extract[:self.to_extract.index(self.extract_until) + 1]
+        self.to_extract_full.append("other")
+        if isinstance(X, pd.DataFrame):
+            self.colnames = X.columns
+        else:
+            self.colnames = None
+        X = check_input(X)
+        self.features_per_column = {}  # Features to extract for each column, after removing constant features
+        for i in range(X.shape[1]):
+            self.features_per_column[i] = []
+        # Check which columns are constant
+        for i in range(X.shape[1]):
+            for feature in self.to_extract_full:
+                if np.nanstd(self._extract_from_date(X[:, i], feature)) > 0:
+                    self.features_per_column[i].append(feature)
+            if self.add_day_of_the_week:
+                self.features_per_column[i].append("dayofweek")
+            if self.add_holidays:
+                self.features_per_column[i].append("holiday")
+
+        self.n_features_out_ = len(np.concatenate(list(self.features_per_column.values())))
+
+        return self
+
+    def transform(self, X, y=None):
+        """ Transform X by replacing each column with corresponding numerical features.
```

Comment:
I think that you are missing an empty line here for this to be valid docstring formatting.